app/app.go

```diff
@@ -132,3 +142,28 @@ var ComposerMap = map[string]Composer{
 	"daily_notes":   compose.DailyNotes,
 	"notes_indexed": compose.NotesIndexed,
 }
+
+func filterModules(array []page.Module, fn func(index int, element page.Module) bool) []page.Module {
+	filtered := make([]page.Module, 0)
+
+	for idx, val := range array {
+		if fn(idx, val) {
+			filtered = append(filtered, val)
+		}
+	}
+
+	return filtered
+}
+
+func uniqueModuleFilter() func(index int, element page.Module) bool {
+	found := map[string]bool{}
+
+	return func(index int, element page.Module) bool {
+		if _, present := found[element.Tpl]; present {
+			return false
+		} else {
+			found[element.Tpl] = true
+			return true
+		}
+	}
+}
```

Comment:
This func is called once directly. Hence, the closure for `found` is redundant.

This construct as a whole looks heavy. What other filters do you anticipate?

Reply:
Removed redundant function in the following commit: https://github.com/kudrykv/latex-yearly-planner/pull/27/commits/6bfe6909f5cc94207914405db31ae5828de13d1c.